hedera-node/src/main/java/com/hedera/services/sigs/verification/PrecheckKeyReqs.java

```diff
@@ -86,7 +87,7 @@ public PrecheckKeyReqs(
 			addQueryPaymentKeys(txn, keys);
 		}
 
-		return keys;
+		return keys.stream().distinct().collect(Collectors.toList());
```

Comment:
Since streams are shockingly expensive, do you think we could, instead, update `addQueryPaymentKeys()` to iterate through the non-payer keys in `otherResult.getOrderedKeys()` and only add a key `nonPayerKey` if `!nonPayerKey.equals(keys.get(0))`?

(This would eliminate the duplication when the `CryptoTransfer` payer and query fee payer are the same; which will be the only duplication 99.99% of the time 😁 ).

Then we could revert this change and avoid the hit from streams.

Reply:
internal duplication could happen when say a token create has multiple custom fees and the fee collectors are same for more than one custom fees.

Reply:
> internal duplication could happen in non-payer keys when say a token create has multiple custom fees and the fee collectors are same for more than one custom fees. But yes this is rare.

A `TokenCreate` txn will never satisfy the `isQueryPayment.test()` tho, right?

Reply:
true